{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running Mr. Bait\n",
    "\n",
    "<br> **Citation:** *Chafin, Tyler and Douglas, Marlis and Douglas, Michael,2018. MrBait: Universal identification and design of targeted-enrichment capture probes.34,Bioinformatics. 10.1093/bioinformatics/bty548*\n",
    "<br> **Description:** A software pipeline for identifying regions of interest in DNA sequence data and designing probes to enrich them.\n",
    "\n",
    "## The workflow is divided into 5 steps, as follows:\n",
    "\n",
    "1. Alignments (provided as .xmfa, .loci, or .maf) or genomes (provided as .fasta, annotated with .vcf or .gff) will be used to build a consensus sequence of each locus.\n",
    "2. A sliding window will be applied to each consensus to find candidate targets for which baits could be designed\n",
    "3. Targets are then selected (if too close together, or only one allowed per locus), and filtered according to any number of specified filter (e.g. GC content, flanking SNPs, pairwise alignment)\n",
    "4. Passing targets are then parsed to design a putative set of baits\n",
    "5. Baits are then filtered according to selected criteria, and output as FASTA.\n",
    "6. The pipeline can be resumed and any steps iteratively re-visited by providing the SQLite database file (resulting in a significant reduction in runtime for successive runs)\n",
    "\n",
    "### Step 1: loading alignments\n",
    "\n",
    "To generate baits of length 80, tiled across target regions with an overlap of 40 bases, from a Multiple Alignment File (MAF) “example.maf”:\n",
    "<br>`mrbait -M example.maf -b 80 -s tile=40`\n",
    "\n",
    "<br>Bait selection scheme: Default behavior is to tile baits across all targets with 50% overlap\n",
    "<br>\n",
    "`mrbait -A ../data/references/Cacao_genomes/Theobroma_cacao_criollo_scaffold.v2.0.fna -T 6 -b 80 -t --strand \"both\" -o scaffTcacao `\n",
    "\n",
    "- `--strand`: Output strand: Use this if you want to print baits as-designed, or as\n",
    "reverse complement. Possible values: “+” [default], “-” (reverse-complement),\n",
    "or “both”. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.1.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bioinformatics/code/scaffTcacao.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:08 (8.600564 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum contig length (-l,--len): 80\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.5\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.5\n",
      "\t\tLoading FASTA file: ../data/references/Cacao_genomes/Theobroma_cacao_criollo_scaffold.v2.0.fna\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 549 loci passed filtering! ###\n",
      "\t\t### Runtime: 0:01:25 (85.172297 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 80\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 0\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 0\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 0\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 500\n",
      "\t\tStarting sliding window target discovery of 549 loci...\n",
      "\t\t\tFinding targets using 6 parallel processes...\n",
      "\n",
      "\t\t### Results: 12192 potential targets identified! ###\n",
      "\t\t### Runtime: 1:05:52 (3952.686254 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): False\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 500\n",
      "\t\tApplying filters to 12192 targets.\n",
      "\t\t\tNo filtering criteria provided. Retaining all targets.\n",
      "\t\tResolving conflicting targets... Found 12192 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Random selection\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 549 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:04 (4.596422 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 80\n",
      "\t\t\tTiling baits with 40 base overlap\n",
      "\n",
      "\t\t### Results: 141697 potential baits identified! ###\n",
      "\t\t### Runtime: 0:04:20 (260.040988 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 141697 potential baits!\n",
      "\t\tApplying filters to 141697 baits...\n",
      "\t\t\tNo filtering criteria provided. Retaining all baits.\n",
      "\t\t\t 141697 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\t\tReporting bait and reverse complement (--strand both)\n",
      "\t\tFinal baits output to: /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bioinformatics/code/scaffTcacao.fasta\n",
      "\n",
      "\t\t### Results: 141697 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:35 (35.822580 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bioinformatics/code/scaffTcacao.fasta ###\n",
      "\t### Runtime: 1:12:26 (4346.920379 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mrbait -A ../data/references/Cacao_genomes/Theobroma_cacao_criollo_scaffold.v2.0.fna -T 6 -b 80 -t --strand \"both\" -o scaffTcacao "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mcacao11genes_pub3i_cds.gz\u001b[0m  \u001b[01;32mTheobroma_cacao_criollo_scaffold.v2.0.fna.tar.gz\u001b[0m\n",
      "\u001b[01;32mReadme.md\u001b[0m                  \u001b[01;32mTheobroma_cacaoV2_annot_cds.fna.tar.gz\u001b[0m\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
